weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/ExternalBeanArchiveProcessor.java

```diff
@@ -325,9 +328,13 @@ private Index tryLoadIndex(URL indexUrl) {
      * @param dependencyModule
      * @return the map of exported resources
      */
-    private Map<URL, URL> findExportedResources(Module dependencyModule) {
+    private Map<URL, URL> findExportedResources(Module dependencyModule, Map<String, Map<URL, URL>> exportedResourcesCache) {
+        if (exportedResourcesCache.containsKey(dependencyModule)) {
```

Comment:
This will always return false as the keys are dependencyModule.getName().

Reply:
Thanks for spotting this @bstansberry! Should be fixed now.